### ISSUE TITLE
Message queue doesn't provide correct message for step finished

### ISSUE THREAD

**Issue by aj062** [buildbot/buildbot#5906]
Message queue doesn't provide correct message for step finished in Buildbot 2.10.1. It used to work in Buildbot 1.8.2.

Below is a code snippet which used to properly print step state_string in Buildbot 1.8.2 and earlier. In Buildbot 2.10.1 (and also in 2.10.2), it returns messages like: "pending", "shell" etc.. For some steps, it prints step "description" instead of "descriptionDone". It seems like this api is invoked by buildbot little too early, before descriptionDone is set.

```class Events(service.BuildbotService):
    @defer.inlineCallbacks
    def startService(self):
        yield service.BuildbotService.startService(self)
        startConsuming = self.master.mq.startConsuming

        self._stepStartedConsumer = yield startConsuming(self.stepStarted, ('steps', None, 'started'))
        self._stepFinishedConsumer = yield startConsuming(self.stepFinished, ('steps', None, 'finished'))

    def stepFinished(self, key, step):
        print('step state_string: {}'.format(step.get('state_string')))
```

**Comment by aj062** [comment]
This was working fine in Buildbot v2.8.4, but stopped working in v2.9.0

cc @tardyp

**Comment by aj062** [comment]
This is a regression from https://github.com/buildbot/buildbot/commit/e2fb0fb3514b99c7e42137a6b97eb6b876a6a97a

Verified that this feature was working fine until this commit.

**Comment by aj062** [comment]
Verified that reverting https://github.com/buildbot/buildbot/commit/e2fb0fb3514b99c7e42137a6b97eb6b876a6a97a fixes this issue.

**Comment by p12tic** [comment]
This is now fixed and released as 2.10.5, 3.0.3 and 3.1.0.
